Source/WebCore/page/Navigation.cpp

```diff
@@ -703,7 +715,8 @@ bool Navigation::innerDispatchNavigateEvent(NavigationNavigationType navigationT
         if (navigationType == NavigationNavigationType::Traverse)
             m_suppressNormalScrollRestorationDuringOngoingNavigation = true;
 
-        // FIXME: Step 7: URL and history update
+        if (navigationType == NavigationNavigationType::Push || navigationType == NavigationNavigationType::Replace)
```

Comment:
navigationTypeToHistoryBehavior may be overkill, since we already narrowed it down to two NavigationNavigationTypes. I'd say we calculate NavigationHistoryBehavior in a local var for now, and if there are more users for it, then navigationTypeToHistoryBehavior starts making sense.